### PR TITLE
Fixes #11583 - Broken HTTP/3 tests

### LINKED ISSUE #11583: Broken HTTP/3 tests
**Jetty version(s)**
12.0.6

**Jetty Environment**
core,ee9,ee10

**Description**
The core distribution test `DistributionTests.testH3` has been boken since 12.0.6 due to #11263, which forgot to add the JNA and Quiche libraries to their own modules.

The ee9 and ee10 tests were broken because of the work done in #8979, because the tests want to explicitly configure the TrustStore, which requires a PEM directory.

### PR DESCRIPTION
**Pull request by sbordet**

### CODE REVIEW

**Comment by lorban** [comment]
@olamy the H3 tests have been broken for some time and we haven't noticed until now because they're skipped on CI. Could you please re-investigate to figure out if they can run reliably now?

**Review by lorban — APPROVED** [review]

File: jetty-core/jetty-http3/jetty-http3-server/src/main/config/modules/jna.mod [hunk 1]
```diff
@@ -5,6 +5,9 @@ https://github.com/java-native-access/jna
 [files]
 maven://net.java.dev.jna/jna-jpms/${jna.version}|lib/http3/jna-jpms-${jna.version}.jar
 
+[lib]
+lib/http3/jna-jpms-${jna.version}.jar
```
Comment (joakime) [inline jetty-core/jetty-http3/jetty-http3-server/src/main/config/modules/jna.mod:9]: ooh.
Should the `setuid.mod` also depend on this `jna` module?

https://github.com/jetty/jetty-setuid-jna/blob/main/src/main/config/modules/setuid.mod

Comment (joakime) [reply jetty-core/jetty-http3/jetty-http3-server/src/main/config/modules/jna.mod:9]: What's the difference between this `jna-jpms-<ver>.jar` and

maven://net.java.dev.jna/jna/${jna.version}|lib/setuid/jna-${jna.version}.jar

?

Comment (joakime) [reply jetty-core/jetty-http3/jetty-http3-server/src/main/config/modules/jna.mod:9]: Should the jpms version only be used for `--jpms` ?

**Comment by olamy** [comment]
> @olamy the H3 tests have been broken for some time and we haven't noticed until now because they're skipped on CI. Could you please re-investigate to figure out if they can run reliably now?

we could re enable them on a branch to see how it's going now.
But except improvement of udp support in k8s I have some doubt...

**Pull request merged**